pulsar-client-cpp/lib/ClientImpl.cc

```diff
@@ -97,20 +97,18 @@ ClientImpl::ClientImpl(const std::string& serviceUrl, const ClientConfiguration&
       requestIdGenerator_(0),
       closingError(ResultOk) {
     std::unique_ptr<LoggerFactory> loggerFactory = clientConfiguration_.impl_->takeLogger();
-    if (!logger) {
```

Comment:
This breaks all custom logging implementations. `loggerFactory` is now set to either `SimpleLoggerFactory` or `Log4CxxLoggerFactory` even if client configuration sets one.